### PR TITLE
Add code freeze bot

### PR DESCRIPTION
**Pull request by gdams**
This GitHub action will enforce the code freeze on repos that have enabled it

### CODE REVIEW

File: .github/workflows/code-freeze.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  freeze:
+    runs-on: ubuntu-latest
+    # Change to false when code freeze is not in place
+    if: true
```
Comment (tellison) [inline .github/workflows/code-freeze.yml:17]: Works for now but would prefer this checkout and test for existence/content of a file in the .github repo to determine global freeze status

File: .github/workflows/code-freeze.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Code Freeze Bot
+on: workflow_call
+
+env:
+  PMC_MEMBERS: '[ "gdams", "sxa", "johnoliver", "tellison", "jerboaa", "smlambert", "karianna", "llxia", "sanhong" ]'
```
Comment (tellison) [inline .github/workflows/code-freeze.yml:7]: Can this be grabbed from Github team membership API?

File: .github/workflows/code-freeze.yml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+  thaw:
+    needs: freeze
+    runs-on: ubuntu-latest
+    if: always() && needs.freeze.result == 'skipped'
+    steps:
+      - uses: khan/pull-request-comment-trigger@v1.1.0
+        id: thaw
+        with:
+          trigger: '/thaw'
```
Comment (tellison) [inline .github/workflows/code-freeze.yml:135]:
```suggestion
          trigger: '/thaw'
          prefix_only: 'true'
```

**Review by tellison — APPROVED** [review]

File: .github/workflows/code-freeze.yml (hunk 2 above)
Comment (gdams) [reply .github/workflows/code-freeze.yml:7]: There is no GitHub team for PMC right now. I had looked at api.eclipse.org but nothing clear

**Pull request merged**